### PR TITLE
Add CephNFS to OLM's clusterserviceversion

### LINKED ISSUE #3826: Ceph NFS CRD is not in the CSV.
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Ceph NFS CRD is not in the CSV.

**Expected behavior:**
Ceph NFS CRD should be in the CSV or rook should detect when a CRD is not present and not start a watch.

### PR DESCRIPTION
**Pull request by rohantmp**
Signed-off-by: Rohan CJ <rohantmp@gmail.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #3826 

**Checklist:**

- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [x] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [x] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [x] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

[skip ci]

### CODE REVIEW

File: cluster/olm/ceph/assemble/metadata-common.yaml [hunk 1]
```diff
@@ -21,6 +21,11 @@ spec:
         version: v1
         displayName: Ceph Object Store User
         description: Represents a Ceph Object Store User.
+      - kind: CephNFS
+        name: cephnfses.ceph.rook.io
+        version: v1
+        displayName: Ceph NFS
+        description: Represents a cluster of Ceph NFS ganesha gateways.
   displayName: Rook-Ceph
```
Comment (leseb) [inline cluster/olm/ceph/assemble/metadata-common.yaml]: Please add:

`replaces: rook-ceph.v1.0.1`

Comment (rohantmp) [reply cluster/olm/ceph/assemble/metadata-common.yaml]: done

**Comment by rohantmp** [comment]
How's this?

**Comment by travisn** [comment]
Should we separate the NFS change from the OLM version update?

**Comment by leseb** [comment]
> Should we separate the NFS change from the OLM version update?

I personally don't mind. Both since we don't run the CI for this, it might be cleaner indeed.

**Comment by BlaineEXE** [comment]
Can we use fewer acronyms for things and spell them out. I'm not certain what CSV is other than comma-separated values.

File: cluster/olm/ceph/assemble/metadata-common.yaml (hunk 1 above)
Comment (rohantmp) [reply cluster/olm/ceph/assemble/metadata-common.yaml]: moved the release stuff to another PR

**Comment by rohantmp** [comment]
OLM is operator lifecycle manager(package manager), and CSV is clusterserviceversion(package manifest) from the  [operator-framework](https://coreos.com/blog/introducing-operator-framework)

**Comment by BlaineEXE** [comment]
Okay. Please spell out acronyms.

**Pull request merged**